### ISSUE TITLE
New config entry to set email name in email FROM field

### ISSUE THREAD

**Issue by sebastianpiskorski** [piwik/piwik#8252]
Current solution allows only to set email address, bout doesn't allow to set its name.

https://github.com/piwik/piwik/blob/2.14.0-rc1/core/Mail.php#L43-L47

This should be more flexible, because changing a translation is not always possible.

**Comment by tsteur** [comment]
Can you give us a bit more information here? When do you want to call it? What is the problem you want to solve?

**Comment by quba** [comment]
To be able to use custom name for scheduled reports sender. It's possible to configure it for password recovery but there's no config option for scheduled reports.

**Comment by sebastianpiskorski** [comment]
@tsteur Please correct me if I'm wrong. Piwik uses class `\Piwik\Mail` to send Mail. And this is the place, where the From and ReplyTo and others fields are set. Theoretically it is set in `\Piwik\Mail::setFrom()` but for real part of it happens in ``\Piwik\Mail::setDefaultFromPiwik()`. And email address can be changed from config, but email alias/name is taken from Translation. It might be good solution for single instance Piwik, but it is impossible to change it for multi instance Piwik, as every instance uses same translations. Also changing ReplyTo field happens somewhere above this class, and there is no global config to change it on the end.

To sum up, I think that there should be not mandatory global config, that allows to change final form of From and Reply to fields. At least some clients requests it.

**Comment by tsteur** [comment]
I understood the technical part but was more interested in the problem that you're trying to solve from a user perspective. So we can think about the best solution. 

From @quba I understood it should be different for scheduled reports? Only there or should the "from" be different everywhere?

> To sum up, I think that there should be not mandatory global config, that allows to change final form of From and Reply to fields. At least some clients requests it.

Here I understand it should be configured and applied everywhere?

It would be often good to know the background... meaning eg what is the problem and how would it "look" when the problem is solved. All rather from a user point of view than technical. 

To sum up. Is it okay if we add a new config where the email alias can be configured?

**Comment by quba** [comment]
In general it should be possible to change the alias in all e-mails that Piwik sends to end users.

>To sum up. Is it okay if we add a new config where the email alias can be configured?
yes, that would be great ;)

**Comment by sebastianpiskorski** [comment]
> To sum up. Is it okay if we add a new config where the email alias can be configured?
:+1:

**Comment by quba** [comment]
great, thanks!
